src/common/session.ts

```diff
@@ -135,3 +123,58 @@ async function writeSessionFieldForSessionId(
         }
     }
 }
+
+type ExpressionAttributeNames = Record<string, string>;
+type ExpressionAttributeValues = Record<string, string | number>;
+type UpdateExpression = string;
+type QueryParameters = {
+    expressionAttributeNames: ExpressionAttributeNames;
+    expressionAttributeValues: ExpressionAttributeValues;
+    updateExpression: UpdateExpression;
+};
+
+/**
+ * Generate DynamoDB update query parameters from a partial session object
+ * @param sessionId Primary key of the record to update
+ * @param partialSession Field name: value pairs to update. Undefined values will be removed
+ *
+ * Given a partial session object, construct SET statements for all present values and REMOVE statements for all
```

Comment:
So does this mean that if the session no longer contains a key/value pair it is removed from the dynamo db entry?

Reply:
Only if the key is explicitly included in the object and the value is `undefined`

Reply:
👍